lib/situation_room/site_controller.ex

```diff
@@ -0,0 +1,79 @@
+defmodule SituationRoom.Site.Controller do
+  alias SituationRoom.Site
+  use Plug.Router
+
+  plug(:match)
+  # If we are going to include plug.Parsers, they go right here
+  plug(:dispatch)
+
+  # Get a site by name
+  get "/name/:name" do
+    get_site_by(conn, name: conn.params["name"])
+  end
+
+  # Get a site by endpoint/url
+  get "/endpoint/:endpoint" do
+    get_site_by(conn, endpoint: conn.params["endpoint"])
+  end
+
+  # Get a site by id
+  get "/id/:id" do
+    get_site_by(conn, id: conn.params["id"])
+  end
+
+  # This route can be hit when forward slashes are encoded as %2F
+  post "/:name/:endpoint" do
```

Comment:
Use restful routes here as well to represent the CRUD operations on a `site`. 

Site creation should be a

   POST /sites

route.  Retrieval should be

    GET /sites
    GET /sites/:id

to list all sites or fetch a single site respectively.